org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/CodeConfig.xtend

```diff
@@ -40,7 +40,7 @@ class CodeConfig implements IGuiceAwareGeneratorComponent {
 	String fileHeaderTemplate = "/*\n * generated by Xtext\n */"
 	
 	@Accessors(PUBLIC_GETTER)
-	val List<IClassAnnotation> classAnnotations = newArrayList
+	val Set<IClassAnnotation> classAnnotations = newLinkedHashSet
```

Comment:
This change is not necessary, since you already do `.toSet` in `GeneratedJavaFileAccess`. The class `CodeConfig` is public API, so we should avoid signature changes if possible.